### PR TITLE
RES-1576: fuse Conv, Batchnorm and ReLU whenever possible

### PR DESCRIPTION
**Pull request by lscheinkman**

### CODE REVIEW

File: nupic/research/frameworks/pytorch/models/resnets.py [hunk 1]
```diff
@@ -149,19 +150,35 @@ def forward(self, x):
...
 
     def fuse_model(self):
-        # Fuse Conv2d and BatchNorm2d modules in regular path
-        cnn_bn = [x[0] for x in self.regular_path.named_modules()
-                  if isinstance(x[1], (nn.Conv2d, nn.BatchNorm2d))]
-        if cnn_bn:
-            modules_to_fuse = list(zip(cnn_bn[::2], cnn_bn[1::2]))
-            fuse_modules(self.regular_path, modules_to_fuse, inplace=True)
+        """
+        Fuse Conv, BathcNorm and ReLU for the first 2 layers and for the 3rd and
```
Comment (mrcslws) [inline nupic/research/frameworks/pytorch/models/resnets.py:154]: Typo: BathcNorm

File: nupic/research/frameworks/pytorch/models/resnets.py [hunk 2]
```diff
@@ -149,19 +150,35 @@ def forward(self, x):
...
+        shortcut lyers.
```
Comment (mrcslws) [inline nupic/research/frameworks/pytorch/models/resnets.py:155]: Typo: lyers

**Comment by lucasosouza** [comment]
Why not merging ReLU on the BasicBlock as well?

**Comment by lscheinkman** [comment]
> Why not merging ReLU on the BasicBlock as well?

We are not using BasicBlock

**Comment by lscheinkman** [comment]
> Why not merging ReLU on the BasicBlock as well?

Done. Test with `resnet34` covers `BasicBlock` and `resnet50` covers the `Bottleneck` block
